src/main/java/com/example/springboot/dao/BookDAOService.java

```diff
@@ -0,0 +1,188 @@
+package com.example.springboot.dao;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Repository;
+
+import com.example.springboot.entity.Book;
+import com.mysql.jdbc.Statement;
+
+/**
+ * This class implement methods from IBookDAOService interface
+ * @author Kohut Dmytro
+ * @version 1.0
+ */
+@Repository
+@ComponentScan
+public class BookDAOService implements IBookDAOService {
+	
+	@Value("${jdbc.url}") private String URL;
+	@Value("${jdbc.username}") private String USERNAME;  
+	@Value("${jdbc.password}") private String PASSWORD;
+	
+	private static final Logger LOGGER = Logger.getLogger(BookDAOService.class);
+	
+	private static final String QUERY_FIND_ALL = "SELECT id, title, author, reader_id FROM `books`;";
+	private static final String QUERY_SELECT_BY_ID = "SELECT id, title, author, reader_id FROM `books` WHERE id=?;";
+	private static final String QUERY_CREATE = "INSERT INTO `books`(title, author, reader_id) VALUES(?, ?, ?);";
+	private static final String QUERY_UPDATE = "UPDATE `books` SET title=?, author=?, reader_id=? WHERE id=?;";
+	private static final String QUERY_DELETE = "DELETE FROM `books` WHERE id=?;";
+	
+	private static final String ID = "id";
+	private static final String TITLE = "title";
+	private static final String AUTHOR = "author";
+	private static final String READER_ID = "reader_id";
+	
+	/**
+	 * This method create and return a connection
+	 * @return Connection
+	 * @throws SQLException 
+	 */
+	private Connection getConnection() throws SQLException {
+		return DriverManager.getConnection(URL, USERNAME, PASSWORD);
+	}
+	
+	/**
+	 * This method create new instance of Book and return it
+	 * @return Book
+	 * @throws SQLException 
+	 */
+	private Book getBook(ResultSet resultSet) throws SQLException {
+		return new Book(resultSet.getInt(ID), resultSet.getString(TITLE), resultSet.getString(AUTHOR), resultSet.getInt(READER_ID));
+	}
+	
+	/**
+	 * This method return all books from database
+	 * @return List<Book>
+	 */
+	@Override
+	public List<Book> findAll() {
+		List<Book> resultList = new ArrayList<>();
+		
+		try (Connection connection = getConnection()) {
+			PreparedStatement statement = connection.prepareStatement(QUERY_FIND_ALL);
+			ResultSet resultSet = statement.executeQuery();
+			
+			while(resultSet.next()) {
+				resultList.add(getBook(resultSet));
+			}
+			
+			return resultList;
+			
+		} catch(SQLException | RuntimeException e) {
+			LOGGER.error(e.getMessage());
```

Comment:
best practice, if we return empty collection throw exception and when we really have exception just log it as error and go by the flow. If DB doesn't work we can return empty list or null and catch all exceptions ;) We even should not inform developer about DB issues. Why all exceptions we are catching, system works stable ;)

Reply:
Instead, e.getMessage () will try to use e, because if you do not use e , Sonar will notify you of a critical error